HW1/P2/P2.txt

```diff
@@ -1 +1,12 @@
 # Your discussion here
+
+As indicated by the questions in P2, the problem with the first method in P2a is the fact that the workload is very badly dispatched over the workers. If one looks at the first histogram, 90% of the tasks are finished much faster than the rest, giving a total computing time of about 30s.
+
+This can be explained by the fact that the computationally expensive tasks are the ones in the middle of the Mandelbrot fractal. Since by default, the rdd is partitioned in sequence 
+(the first 100 values, the second 100, etc), we are left with only black partitions and only white partitions which are much slower to evaluate.
+
+We therefore need a better way to balance the workload by creating partitions that have both black and white pixels in them. I first wanted to create partitions by explicitly adding black and white pixels with the same amounts for the 100 partitions. But this was too complicated since it was difficult to actually get the coordinates of these two categories, and because of the x-axis asymetry.  
```

Comment:
I once thought about this method too. I believe this method is really hard to achieve.